search-extension/src/main/java/com/afrigis/services/search/extension/intiendols/api2/params/CensusParams.java

```diff
@@ -0,0 +1,219 @@
+     * </p>
+     *
+     * @return String
+     */
+    public String getLongitude() {
+        return longitude;
+    }
+
+    /**
+     * <p>
+     * Set longitude to identify location
+     * </p>
+     *
+     * @param longitude
+     */
+    public void setLongitude(String longitude) {
+        this.longitude = longitude;
+    }
+
+    /**
+     * <p>
+     * Get Census report using seoid
+     * </p>
+     *
+     * @param email AfriGIS service client email making service call
+     * @param seoid location identifier
+     * @param censusGetType get JSON or PDF report
+     */
+    public CensusParams(String email, String seoid, CensusGetType censusGetType) {
+        this.email = email;
+        this.seoid = seoid;
+        this.censusGetType = censusGetType;
+    }
+
+    /**
+     *
+     * <p>
+     * Get Census report using latitude and longitude
+     * </p>
+     *
+     * @param email AfriGIS service client email making service call
+     * @param latitude identify location
+     * @param longitude identify location
+     * @param censusGetType get JSON or PDF report
+     */
+    public CensusParams(String email, String latitude, String longitude, CensusGetType censusGetType) {
+        this.latitude = latitude;
+        this.longitude = longitude;
+        this.email = email;
+        this.censusGetType = censusGetType;
+    }
+
+    /**
+     * <p>
+     * Add AfriGIS service specific parameters
+     * </p>
+     *
+     * @param input
+     */
+    @Override
+    protected void completeRequestParamList(Collection<KeyValue> input) {
+        addKeyValParam(EMAIL, getEmail(), input);
+
+        if (getSeoid() != null && getSeoid().trim().length() > 0) {
+            addKeyValParam(SEOID, getSeoid(), input);
+        } else {
+            addKeyValParam(LATITUDE, getLatitude(), input);
+            addKeyValParam(LONGITUDE, getLongitude(), input);
+        }
+    }
+
+    /**
+     * <p>
+     * Service to call, this based on censusGetType
+     * </p>
+     *
+     * @return String service to call
+     */
+    @Override
+    public String getServiceName() {
+        return censusGetType == CensusGetType.JSON ? "intiendols.extn.census" : "intiendols.extn.census.download";
+    }
+
+    /**
+     * <p>
+     * Verify all required service parameters and their format
+     * </p>
+     *
+     * @throws AfriGISServicesException
+     */
+    @Override
+    public void validate() throws AfriGISServicesException {
+        if (StringUtils.isEmpty(getEmail())) {
+            throw new AfriGISServicesException(
+                    "Exceptions - missing mandatory parameter (email)");
+        }
+        // Valid email?
+        if (!getEmail().matches(EMAIL_REGEX)) {
+            throw new AfriGISServicesException(
+                    "Exceptions - invalid email");
+        }
+
+        if (StringUtils.isEmpty(getSeoid()) && (StringUtils.isEmpty(getLatitude()) || StringUtils.isEmpty(getLongitude()))) {
```

Comment:
In general, consider using `isBlank` if available, since it also protects you against input like `       `